packages/react-percy-ci/src/Environment.js

```diff
@@ -1,41 +1,26 @@
+import { evalVMScript, jsdom } from 'jsdom';
 import createSuite from '@percy-io/react-percy-test-framework';
 import vm from 'vm';
 
-const GLOBALS = [
-  'clearImmediate',
-  'clearInterval',
-  'clearTimeout',
-  'console',
-  'setImmediate',
-  'setInterval',
-  'setTimeout',
-];
-
 export default class Environment {
-  constructor(percyConfig) {
-    this.percyConfig = percyConfig;
-    this.context = vm.createContext();
-    this.global = vm.runInContext('this', this.context);
-    this.global.global = this.global;
-    GLOBALS.forEach(key => {
-      this.global[key] = global[key];
-    });
-    this.suite = createSuite(this.global);
+  constructor() {
+    this.context = {};
+    this.rootSuite = createSuite(this.context);
   }
 
-  getSnapshots() {
-    return this.suite.getSnapshots();
+  getSnapshotDefinitions() {
+    return this.rootSuite.getSnapshotDefinitions();
```

Comment:
Server-side we only need to get the names and options for all the defined snapshots, we don't actually need to run the markup-generating functions for each.